project/src/main/puzzle/critter/critter-manager.gd

```diff
@@ -149,7 +149,8 @@ func update_piece_manager_piece(new_type: PieceType, new_pos: Vector2, new_orien
 ## If all blocks were removed from a piece while being moved, we manually apply some steps which usually happen
 ## automatically such as pausing and running certain triggers. This occurs when sharks eat an entire piece.
 func check_for_empty_piece() -> void:
-	if _piece_manager.piece.type.empty() and _piece_manager.get_state() == _piece_manager.states.move_piece:
+	if _piece_manager.piece.type.empty() \
+			and _piece_manager.get_state() in [_piece_manager.states.move_piece, _piece_manager.states.prespawn]:
```

Comment:
Maybe comment why 'prespawn' is included here

Reply:
Comment; `A piece can spawn directly on a blue shark and disappear before entering the move_piece state...`